rubicon_ml/client/artifact.py

```diff
@@ -40,10 +41,18 @@ def __init__(self, domain, parent):
     def _get_data(self):
         """Loads the data associated with this artifact."""
         project_name, experiment_id = self.parent._get_identifiers()
-
-        self._data = self.repository.get_artifact_data(
-            project_name, self.id, experiment_id=experiment_id
-        )
+        for repo in self.repositories:
+            self._data = None
+            try:
+                self._data = repo.get_artifact_data(
+                    project_name, self.id, experiment_id=experiment_id
+                )
+            except Exception as err:
+                return_err = err
+            else:
+                return
+        if self._data is None:
+            return RubiconException(return_err)
```

Comment:
this needs to be a `raise`

Reply:
can we also change all of the `raise RubiconException(return_err)` to `raise RubiconException("all configured storage backends failed") from return_err`?

using the `from` syntax will preserve the original error's callback when its printed, which should be easier for users to debug